File: backend/src/main/java/com/fateczl/sistemaDeGestaoEscolar/academico/matriz/MatrizCurricular.java
package com.fateczl.sistemaDeGestaoEscolar.academico.matriz;

import com.fateczl.sistemaDeGestaoEscolar.academico.StatusMatriz;
import com.fateczl.sistemaDeGestaoEscolar.disciplina.Disciplina;
import com.fateczl.sistemaDeGestaoEscolar.turma.Turma;
import com.fateczl.sistemaDeGestaoEscolar.usuario.funcionario.Funcionario;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "matriz_curricular",
uniqueConstraints = @UniqueConstraint(
    name = "uk_matriz_turma_disciplina_ano",
    columnNames = {"turma_id", "disciplina_id", "ano"}))
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(of = "id")
public class MatrizCurricular {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "turma_id", nullable = false)
    private Turma turma;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "disciplina_id", nullable = false)
    private Disciplina disciplina;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "professor_id", nullable = false)
    private Funcionario professor;

    @Column(nullable = false)
    private int ano;

    @Column(nullable = false)
    private int cargaHorariaTotal;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusMatriz status = StatusMatriz.ATIVA;

    
}

File: frontend/lib/models/matriz_curricular.dart
class MatrizCurricular {
  final int id;
  final int turmaId;
  final String nomeTurma;
  final int disciplinaId;
  final String nomeDisciplina;
  final int professorId;
  final String nomeProfessor;
  final int ano;
  final int cargaHorariaTotal;
  final int aulasRealizadas;
  final String status;

  MatrizCurricular({
    required this.id,
    required this.turmaId,
    required this.nomeTurma,
    required this.disciplinaId,
    required this.nomeDisciplina,
    required this.professorId,
    required this.nomeProfessor,
    required this.ano,
    required this.cargaHorariaTotal,
    required this.aulasRealizadas,
    required this.status,
  });

  factory MatrizCurricular.fromJson(Map<String, dynamic> j) => MatrizCurricular(
    id: j['id'],
    turmaId: j['turmaId'],
    nomeTurma: j['nomeTurma'],
    disciplinaId: j['disciplinaId'],
    nomeDisciplina: j['nomeDisciplina'],
    professorId: j['professorId'],
    nomeProfessor: j['nomeProfessor'],
    ano: j['ano'],
    cargaHorariaTotal: j['cargaHorariaTotal'],
    aulasRealizadas: j['aulasRealizadas'],
    status: j['status'],
  );
}
